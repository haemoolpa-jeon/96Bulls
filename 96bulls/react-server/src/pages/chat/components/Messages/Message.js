import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Message.module.scss';

class Message extends Component {
  isImage = message => message.hasOwnProperty('image') && !message.hasOwnProperty('content');

  render() {
    const { message, user } = this.props;
    return (
      <div className={styles.message}>
        <img className={styles.avatar} src={user.avatar} alt={`${user.name} avatar`} />
        <div className={styles['message-content']}>
          <span className={styles['display-name']}>{user.name}</span>
          {this.isImage(message)
            ? (
              <div className={styles['image-wrapper']}>
                <img
                  className={styles['message-img']}
                  src={message.image}
                  alt={message.image}
                />
              </div>
            )
            : <p className={styles['message-text']}>{message.content}</p>
          }
        </div>
      </div>
    );
  }
}

Message.propTypes = {
  message: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
};

export default Message;
