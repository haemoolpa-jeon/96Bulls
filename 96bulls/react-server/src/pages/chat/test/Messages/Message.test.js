import React from 'react';
import Message from '../../components/Messages/Message';
import {shallow} from 'enzyme/build';

describe('Message', () => {
  let component = null;

  const user = {
    avatar: 'http://wwww.mock-image-url1.com',
    name: 'myp1',
    email: 'myp1@daum.net'
  };

  const imgMessage = {
    image: 'www.image.net'
  };

  const contentMessage = {
    content: '안녕하세요'
  }

  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<Message
      message={contentMessage}
      user={user}
    />);
  });

  it('이미지 메세지가 전달되면 img 태그를 랜더링 함', () => {
    component = shallow(<Message
      message={imgMessage}
      user={user}
    />);
    component.find('img').exists()
  });


  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });
});
