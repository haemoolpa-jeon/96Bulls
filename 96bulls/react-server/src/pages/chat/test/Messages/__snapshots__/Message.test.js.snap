// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message 초기 렌더링 스냅샷 일치함 1`] = `
<div
  className="message"
>
  <img
    alt="myp1 avatar"
    className="avatar"
    src="http://wwww.mock-image-url1.com"
  />
  <div
    className="message-content"
  >
    <span
      className="display-name"
    >
      myp1
    </span>
    <div
      className="image-wrapper"
    >
      <img
        alt="www.image.net"
        className="message-img"
        src="www.image.net"
      />
    </div>
  </div>
</div>
`;
